app/models/package_manager/packagist.rb

```diff
@@ -97,9 +97,9 @@ def self.versions(raw_project, _name)
     end
 
     def self.acceptable_versions(versions)
-      versions.select do |k, _v|
+      versions.select do |v|
         # See: https://getcomposer.org/doc/articles/versions.md#branches
-        (k =~ /^dev-.*/i).nil? && (k =~ /\.x-dev$/i).nil?
+        (v["version"] =~ /^dev-.*/i).nil? && (v["version"] =~ /\.x-dev$/i).nil?
```

Comment:
this was a bug that was hidden by the fact that `Hash#respond_to?(:=~)`, but it was caught during this upgrade because `Hash#respond_to?(:=~)` is no longer true.